WooCommerce/Classes/POS/Presentation/PointOfSaleCollectCashView.swift

```diff
@@ -70,16 +65,7 @@ struct PointOfSaleCollectCashView: View {
 
             Button(action: {
                 Task { @MainActor in
-                    guard validateAmountOnSubmit() else {
-                        return
-                    }
-                    isLoading = true
-                    do {
-                        try await markComplete()
-                    } catch {
-                        errorMessage = Localization.failedToCollectCashPayment
-                    }
-                    isLoading = false
+                    await submitCashAmount()
```

Comment:
There's also a case where the keyboard starts hiding the button. We could also check with iPad Mini, the keyboard may be hiding the whole button in some cases.

In `UIKit` days, we used to create reusable functionality by tracking the keyboard size and shrinking the view in the process. Maye there's a recommendation of a similar `SwiftUI` modifier.  Usually in these scenarios scroll views were used. In this case, we have a a header, text field, and a button. We should be able to fit the elements without using a scroll view.

![IMG_0157](https://github.com/user-attachments/assets/3ed62778-def4-4c95-a015-30fcbcd61447)

Reply:
> There's also a case where the keyboard starts hiding the button. We could also check with iPad Mini, the keyboard may be hiding the whole button in some cases.

That was the case indeed. In the case of iPad mini I'm setting the padding to 0 for all cases so it fits nicely. [The device check](https://github.com/woocommerce/woocommerce-ios/blob/6b52c7c3aa4c0f1d0f9df4056125b8462e5a6448/WooCommerce/Classes/POS/Presentation/TotalsView.swift#L398) feels a bit like a hack thought.

![_ipad_mini](https://github.com/user-attachments/assets/2b200fa6-bb7b-4899-af4c-8faa99a58fab)

Reply:
> [The device check](https://github.com/woocommerce/woocommerce-ios/blob/6b52c7c3aa4c0f1d0f9df4056125b8462e5a6448/WooCommerce/Classes/POS/Presentation/TotalsView.swift#L398) feels a bit like a hack thought.

Yes, it does. We should try to avoid it.

The way of thinking could be a keyboard acting as an element that pushes the button up. Spaces would shrink when the keyboard pushes button. Spacers could have a max height so the view wouldn't scatter too much. 

I used to do it like that with UIKit but haven't done with SwiftUI. So I hope there _should_ be the way.

Header
Spacer
Text Field
Spacer
Button
Spacer
Keyboard